packages/account/imAccount/index.ts

```diff
@@ -0,0 +1,119 @@
+import * as ethers from "ethers"
+
+import type { Call } from "~packages/account"
+import imAccountMetadata from "~packages/account/imAccount/abis/imAccount.json"
+import type { Validator } from "~packages/account/imAccount/validator"
+import { AccountSkeleton } from "~packages/account/skeleton"
+import type { BigNumberish, HexString } from "~typing"
+
+import { imAccountFactory } from "./factory"
+
+export class imAccount extends AccountSkeleton<imAccountFactory> {
+  /**
+   * Use when account is already deployed
+   */
+  public static async init(opts: {
+    address: string
+    nodeRpcUrl: string
+    validator: Validator
+    entryPointAddress: string
+  }) {
+    return new imAccount({
+      address: opts.address,
+      nodeRpcUrl: opts.nodeRpcUrl,
+      validator: opts.validator,
+      entryPointAddress: opts.entryPointAddress
+    })
+  }
+
+  /**
+   * Use when account is not yet deployed
+   */
+
+  public static async initWithFactory(opts: {
+    factoryAddress: string
+    implementationAddress: string
+    entryPointAddress: string
+    validator: Validator
+    fallbackHandlerAddress: string
+    salt: BigNumberish
+    nodeRpcUrl: string
+  }) {
+    const factory = new imAccountFactory({
+      factoryAddress: opts.factoryAddress,
+      implementationAddress: opts.implementationAddress,
+      entryPointAddress: opts.entryPointAddress,
+      validator: opts.validator,
+      fallbackHandlerAddress: opts.fallbackHandlerAddress,
+      salt: opts.salt,
+      nodeRpcUrl: opts.nodeRpcUrl
+    })
+    return new imAccount({
+      address: await factory.getAddress(),
+      factory,
+      nodeRpcUrl: opts.nodeRpcUrl,
+      validator: opts.validator,
+      entryPointAddress: opts.entryPointAddress
+    })
+  }
+
+  private account: ethers.Contract
+  private validator: Validator
+  private entryPointAddress: string
+
+  private constructor(opts: {
+    address: HexString
+    factory?: imAccountFactory
+    nodeRpcUrl: string
+    validator: Validator
+    entryPointAddress: string
+  }) {
+    super({
+      address: opts.address,
+      factory: opts.factory,
+      nodeRpcUrl: opts.nodeRpcUrl
+    })
+    this.account = new ethers.Contract(
+      opts.address,
+      imAccountMetadata.abi,
+      this.node
+    )
+    this.validator = opts.validator
+    this.entryPointAddress = opts.entryPointAddress
+  }
+
+  public changeValidator(newValidator: Validator) {
+    this.validator = newValidator
+  }
+
+  public async sign(message: string | Uint8Array, metadata?: any) {
```

Comment:
nit: We can leverage the type `BytesLike`

```suggestion
import type { BytesLike } from "~typing"
// ...
  public async sign(message: BytesLike, metadata?: any) {
```